fireapp/lib/pages/settings/setting.dart

```diff
@@ -92,14 +93,14 @@ class _SettingsState extends State<SettingBox> {
                     );
                   }),
               SettingsTile.navigation(
-                leading: const Icon(Icons.logout),
-                title: const Text('Sign Out'),
-                onPressed: (_) {
-                  _authenticationRepository.logout();
-                  Navigator.popUntil(context, (route) => true);
-                  Navigator.pushNamed(context, "/login");
-                }
-              ),
+                  leading: const Icon(Icons.logout),
+                  title: const Text('Sign Out'),
+                  onPressed: (_) async {
+                    await viewModel.unregisterFcmToken();
```

Comment:
Everything in this function is business logic, put it in the view model

Reply:
The function was there before. Should I extract the entire function and put it in the ViewModel? I'm not quite sure about the navigation part.

Reply:
Yes, extract all of it. The navigation can be handled the same way as in the login view model/view